docs/guides/extensions/panels-display-data-vonage.md

```diff
@@ -0,0 +1,769 @@
+				{ text: 'Deleted', value: 'deleted' }
+			]
+		}
+	}
+},
+```
+
+After the `options` section, there is the ability to limit the width and height of the panel. Since this panel will hold
+a lot of data, set these to `24` for the width and `18` for the height:
+
+```js
+minWidth: 24,
+minHeight: 18,
+```
+
+The output of these options will look like this:
+
+<img src="https://marketing.directus.app/assets/053298df-a5ff-4fd5-85ec-ff0f78bb64ca.png" alt="Form shows product type dropdown, direction dropdown, date range dropdown, included message checkbox, and status dropdown." style="padding: 2px 12px;">
+
+## Prepare the View
+
+Open the `panel.vue` file and you will see the starter template and script. Skip to the script section and import the
+following packages:
+
+```js
+import { useApi } from '@directus/extensions-sdk';
+import { adjustDate } from '@directus/shared/utils';
+import { formatISO, formatDistanceToNow, parseISO } from 'date-fns';
+import { ref, watch } from 'vue';
+```
+
+In the `props`, `showHeader` is one of the built-in properties which you can use to alter your panel if a header is
+showing. Remove the text property and add all the options that were created in the previous file:
+
+```js
+props: {
+	showHeader: {
+		type: Boolean,
+		default: false,
+	},
+	type: {
+		type: String,
+		default: '',
+	},
+	direction: {
+		type: String,
+		default: '',
+	},
+	range: {
+		type: String,
+		default: '',
+	},
+	include_message: {
+		type: Boolean,
+		default: false,
+	},
+	status: {
+		type: String,
+		default: '',
+	},
+},
+```
+
+After the `props`, create a `setup(props)` section and create the variables needed:
+
+```js
+
+setup(props){
+	const api = useApi();
+	const activity_data = ref([]);
+	const now = ref(new Date());
+	const is_loading = ref(true);
+	const has_error = ref(false);
+},
+
+```
+
+Create a `fetchData` function that will use the information provided to construct the query parameters and perform the
+API query. The response is written to the `activity_data` constant.
+
+Use the `is_loading` constant to hide or show the progress spinner to indicate that the query is running:
+
+```js
+async function fetchData(){
+	is_loading.value = true;
+	activity_data.value = [];
+
+	const params = {
+		product: props.type || 'SMS',
+		direction: props.direction || 'outbound',
+		include_message: props.include_message || 'false',
+		date_start: formatISO(adjustDate(now.value, (props.range ? `-${props.range}` : '-1 day'))),
+		status: props.status || 'any'
+	};
+	if(props.status) params.status = props.status;
+
+	let url_params = new URLSearchParams(params);
+	try {
+		api.get(`/vonage/records?${url_params.toString()}`).then((rsp) => {
+			activity_data.value = rsp.data.records;
+			is_loading.value = false;
+		}).catch((error) => {
+			is_loading.value = false;
+			has_error.value = error.message;
+		});
+	} catch (err) {
+		is_loading.value = false;
+		has_error.value = "Internal Server Error";
+		console.warn(err);
+	}
```

Comment:
This try/catch block doesn't do anything, as the promise already has it's own `.catch` error handler

```suggestion
	try {
		const rsp = await api.get(`/vonage/records?${url_params.toString()}`)
		activity_data.value = rsp.data.records;
	} catch (err) {
		has_error.value = "Internal Server Error";
		console.warn(err);
	} finally {
		is_loading.value = false;
	}
```